### PR TITLE
Add image suffix for kube-apiserver

### PR DESCRIPTION
**Pull request by MarcelMue**
## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: pkg/template/release.go [hunk 1]
```diff
@@ -9,17 +9,17 @@ import (
...
-		KubeProxy:                    buildImage("giantswarm/kube-proxy", versions.Kubernetes),
-		KubeScheduler:                buildImage("giantswarm/kube-scheduler", versions.Kubernetes),
-		KubernetesAPIHealthz:         buildImage("giantswarm/k8s-api-healthz", versions.KubernetesAPIHealthz),
-		KubernetesNetworkSetupDocker: buildImage("giantswarm/k8s-setup-network-environment", versions.KubernetesNetworkSetupDocker),
+		CalicoCNI:                    buildImage("giantswarm/cni", versions.Calico, ""),
+		CalicoKubeControllers:        buildImage("giantswarm/kube-controllers", versions.Calico, ""),
+		CalicoNode:                   buildImage("giantswarm/node", versions.Calico, ""),
+		Etcd:                         buildImage("giantswarm/etcd", versions.Etcd, ""),
+		Hyperkube:                    buildImage("giantswarm/hyperkube", versions.Kubernetes, ""),
+		KubeApiserver:                buildImage("giantswarm/kube-apiserver", versions.Kubernetes, "-giantswarm"),
```
Comment (corest) [inline pkg/template/release.go:17]: I've looked into quay and didn't find this repo

Comment (tfussell) [reply pkg/template/release.go:17]: kube-apiserver? It's here https://quay.io/repository/giantswarm/kube-apiserver

Comment (corest) [reply pkg/template/release.go:17]: Ah, crap. `-giantswarm` is a tag prefix, not a repo prefix

**Comment by MarcelMue** [comment]
@giantswarm/team-ludacris PTAL - this was tested successfully yesterday.

File: pkg/template/release.go [hunk 2]
```diff
@@ -9,17 +9,17 @@ import (
...
-		KubeProxy:                    buildImage(registryDomain, "giantswarm/kube-proxy", versions.Kubernetes),
-		KubeScheduler:                buildImage(registryDomain, "giantswarm/kube-scheduler", versions.Kubernetes),
-		KubernetesAPIHealthz:         buildImage(registryDomain, "giantswarm/k8s-api-healthz", versions.KubernetesAPIHealthz),
-		KubernetesNetworkSetupDocker: buildImage(registryDomain, "giantswarm/k8s-setup-network-environment", versions.KubernetesNetworkSetupDocker),
+		CalicoCNI:                    buildImage(registryDomain, "giantswarm/cni", versions.Calico, ""),
+		CalicoKubeControllers:        buildImage(registryDomain, "giantswarm/kube-controllers", versions.Calico, ""),
+		CalicoNode:                   buildImage(registryDomain, "giantswarm/node", versions.Calico, ""),
+		Etcd:                         buildImage(registryDomain, "giantswarm/etcd", versions.Etcd, ""),
+		Hyperkube:                    buildImage(registryDomain, "giantswarm/hyperkube", versions.Kubernetes, ""),
+		KubeApiserver:                buildImage(registryDomain, "giantswarm/kube-apiserver", versions.Kubernetes, "-giantswarm"),
```
Comment (stone-z) [inline pkg/template/release.go:17]: Not a huge fan of hardcoding this but let's get it out

File: pkg/template/release.go [hunk 3]
```diff
@@ -59,13 +59,13 @@ func ExtractComponentVersions(releaseComponents []v1alpha1.ReleaseSpecComponent)
...
 }
 
-func buildImage(registryDomain, repo, tag string) string {
+func buildImage(registryDomain, repo, tag, suffix string) string {
 	name := repo
 	if registryDomain != "" {
 		name = registryDomain + "/" + repo
 	}
 
-	return name + ":" + tag
```
Comment (tfussell) [inline pkg/template/release.go:68]: Paweł just changed this from `fmt.Sprintf` to direct string appending I think. Should we prefer one or the other?

Comment (MarcelMue) [reply pkg/template/release.go]: I am a fan of `fmt` - just committed my change.

**Pull request merged**